automerge/src/automerge.rs

```diff
@@ -310,7 +310,7 @@ impl Automerge {
     /// For a list the keys are the element ids (opids) encoded as strings.
     pub fn values<O: AsRef<ExId>>(&self, obj: O) -> Values<'_> {
         if let Ok(obj) = self.exid_to_obj(obj.as_ref()) {
-            let iter_range = self.ops.range(obj, ..);
+            let iter_range = self.ops.list_range(obj, ..);
```

Comment:
Does this now remove support for `values` on a map?

Reply:
Did you want something like this?

```
#[derive(Debug)]
pub(crate) enum ValuesIter<'a> {
    None,
    List(query::ListRange<'a, RangeFull>),
    Map(query::MapRange<'a, RangeFull>),
}


#[derive(Debug)]
pub struct Values<'a> {
    range: ValuesIter<'a>,
    doc: &'a Automerge,
}

impl<'a> Values<'a> {
    pub(crate) fn new<T: Into<ValuesIter<'a>>>(doc: &'a Automerge, range: T) -> Self {
        Self {
            range: range.into(),
            doc,
        }
    }
}

impl<'a> Iterator for Values<'a> {
    type Item = (Value<'a>, ExId);

    fn next(&mut self) -> Option<Self::Item> {
        match &mut self.range {
            ValuesIter::None => None,
            ValuesIter::List(l) => l
                .next()
                .map(|(_idx, value, id)| (value, self.doc.id_to_exid(id))),
            ValuesIter::Map(m) => m
                .next()
                .map(|(_prop, value, id)| (value, self.doc.id_to_exid(id))),
        }
    }
}```